test/etherscan.spec.js

```diff
@@ -0,0 +1,40 @@
+
+const path = require('path');
+
+describe('GetABI', () => {
+  const address = '0xda7c27c04f66842faf20644814b644e25e1766ea';
+  const wrongAddress = '0xda7c27c04f66842faf20644814b644e25e1766eb';
+  const dirpath = path.join(__dirname, '..', 'contracts');
+  const jsonPath = path.join(dirpath, `${address}.json`);
+
+  beforeEach(() => {
+    rimraf.sync(dirpath);
+  });
+  it('should create a contract directory if there was non', async () => {
+    await getABI(address);
+    expect(fs.existsSync(dirpath)).to.be.equal(true);
+  });
+  it('should scrape smart-contract ABI from etherscan and store locally', async () => {
+    const EtherscanABi = await getABI(address);
+    expect(await EtherscanABi).to.deep.equal(TestABI);
+    expect(JSON.parse(fs.readFileSync(jsonPath, { encoding: 'utf8' }))).to.deep.equal(TestABI);
+  });
+  it('should take ABI from local file when present', async () => {
+    await getABI(address);
+    expect(JSON.parse(fs.readFileSync(jsonPath, { encoding: 'utf8' }))).to.deep.equal(await getABI(address));
+  });
+  it('should throw if etherscan returns a failed REST response', async () => {
+    getABI(wrongAddress).should.be.rejectedWith('Wrong response from Etherscan or wrong Contract Address');
+  });
+});
```

Comment:
We agreed that we will take this test file off, and make it in separate PR later.